packages/jest-mock/src/index.ts

```diff
@@ -725,37 +880,39 @@ class ModuleMockerClass {
       metadata.value = component;
       return metadata;
     } else if (type === 'function') {
+      // @ts-ignore this is a function so it has a name
       metadata.name = component.name;
+      // @ts-ignore may be a mock
       if (component._isMockFunction === true) {
+        // @ts-ignore may be a mock
         metadata.mockImpl = component.getMockImplementation();
       }
     }
 
     metadata.refID = refs.size;
     refs.set(component, metadata.refID);
 
-    let members = null;
+    let members: {[key: string]: MockFunctionMetadata<T, Y>} | null = null;
     // Leave arrays alone
     if (type !== 'array') {
-      if (type !== 'undefined') {
```

Comment:
we check and bail on `'undefined'` earlier so TS told me I can remove it

Reply:
🎉